### DISCUSSION TITLE
Windows authentication implementation

### DISCUSSION THREAD

**Discussion by Rodien** [OrchardCMS/OrchardCore#15654] [Q&A]
Hi everyone, before I start looking into implementing Windows authentication (Active Directory) , is there a community implementation that you all are aware of?

**Comment by Rodien** [comment]
I would like to have the same functionalities as the Microsoft Entra ID module. Users should be able to link to an existing account, or an account should be added to Orchard Core automatically if the user account does not exist.

I am currently looking at the Microsoft Entra ID module, but I cannot figure out when or where the user gets created or linked to an existing account.

Is there anyone in the community who can shed light on this.

**Comment by hishamco** [comment]
I already implemented LDAP a few years ago, but I need to port it into OC

Related to https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/issues/29

**Reply by Rodien** [reply]
Thank you for responding @hishamco . Will you have time to port your LDAP module anytime soon?

**Reply by hishamco** [reply]
Hopefully, I have many things on my list at the moment, but I will struggle to revisit all the issues in OCC after a week
